tests/compile-fail/non_extern_foreign_call_bad_unwind1.rs

```diff
@@ -0,0 +1,12 @@
+#[no_mangle]
+extern "C" fn unwind() {
+    panic!();
+}
```

Comment:
I guess this is one of the tests that won't work any more with `feature(c_unwind)`? Any idea what to do once that feature is stabilized?

Reply:
We could... add a `Machine::enforce_abi_check()` method and a `-Zmiri` flag to disable ABI checks, similar to `-Zmiri-disable-alignment-check`, `-Zmiri-symbolic-alignment-check` and `-Zmiri-disable-validation`? Then we can unwind in a `"C-unwind"` function and call the function using the `"C"`-without-unwind ABI.

Reply:
Hm, yeah that sounds plausible. Nice idea!
(I would also have accepted "doing whatever I did with the top_of_stack test", but of course that would have reduced our test coverage.)